extensions/amp-story/1.0/amp-story.js

```diff
@@ -1284,17 +1328,53 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
-    if (
-      isExperimentOn(this.win, 'amp-story-paywall-exp') &&
-      targetPage.isPaywallProtected()
-    ) {
-      if (this.storeService_.get(StateProperty.SUBSCRIPTIONS_DIALOG_STATE)) {
-        // Subscription dialog is already triggered.
-        return Promise.resolve();
-      }
-      this.storeService_.dispatch(Action.TOGGLE_SUBSCRIPTIONS_DIALOG, true);
+    const subscriptionsState = this.storeService_.get(
+      StateProperty.SUBSCRIPTIONS_STATE
+    );
+    if (this.isPaywallStory_()) {
+      if (
+        pageIndex >= PAYWALL_PAGE_INDEX &&
+        subscriptionsState !== SubscriptionsState.GRANTED
+      ) {
+        if (subscriptionsState === SubscriptionsState.UNKNOWN) {
+          return this.blockOnPendingSubscriptionsState_(
+            targetPageId,
+            direction
+          );
+        }
+
+        // Attempt to navigate to the locked pages after the paywall page.
+        if (pageIndex > PAYWALL_PAGE_INDEX) {
```

Comment:
Re: Directly line 1134 and not nested in any other `if`: `if (SubscriptionsState.BLOCKED && pageIndex > PAYWALL_PAGE_INDEX)`